ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -60,6 +62,19 @@ import           Ouroboros.Network.Subscription.Dns
 import           Ouroboros.Network.Subscription.Worker (LocalAddresses (..))
 import           Ouroboros.Network.Tracers
 
+data DiffusionInitializationTracer
+  = DiffusionInitiatilizationRunServer
+  | DiffusionInitiatilizationRunLocalServer
+  | DiffusionInitiatilizationCreatingLocalServerSocketForUnixPath FilePath
```

Comment:
^ I would call this `SystemdSocket`

Reply:
All fields in this type should be strict.

Reply:
BTW, `type FilePath = String = [Char]`, so adding a bang won't guarantee that there are no thunks, it will only force the first `(:)` constructor, but *not* its arguments.

Reply:
For `FilePath` this is ok, as it will be evaluated by the network library; But this is the reason why turning other `String`s to `CInt` and use `socketToFd` to obtain the file descriptor number, rather than use `show socket` (which would never be evaluated by our code).